packages/core/utils/workspace.ts

```diff
@@ -76,7 +78,19 @@ export async function populateWorkspaceDetails(
 
 	const { data: packageJson } = await getPackageJson(workspace);
 	if (packageJson.devDependencies) {
-		workspace.typescript.installed = 'tslib' in packageJson.devDependencies;
+		const tsConfigFileName = 'tsconfig.json';
+		const viteConfigFileName = 'vite.config.ts';
+		let usesTypescript = await fileExists(path.join(workingDirectory, viteConfigFileName));
+
+		if (remoteControl.isRemoteControlled()) {
+			// while executing tests, we only look into the direct `workingDirectory`
+			// as we might detect the monorepo `tsconfig.json` otherwise.
+			usesTypescript ||= await fileExists(path.join(workingDirectory, tsConfigFileName));
+		} else {
+			usesTypescript ||= await findUp(workingDirectory, tsConfigFileName);
+		}
+
+		workspace.typescript.installed = usesTypescript;
```

Comment:
taking another look at this, we should probably move this out of the `if(packageJson.devDependencies)` check since it doesn't depend on it anymore

Reply:
Good catch